devices/power.py

```diff
@@ -350,6 +350,35 @@ def reset(self):
         time.sleep(5)
         self.on()
 
+class Ip9820(PowerDevice):
+    def __init__(self, ip_address, port, username="admin", password="12345678"):
+        PowerDevice.__init__(self, ip_address, username, password)
+        self._ip_address = ip_address
+        self.port = port
+
+        # create a password manager
+        password_mgr = urllib2.HTTPPasswordMgrWithDefaultRealm()
+        password_mgr.add_password(None, 'http://' + ip_address, username, password)
+        handler = urllib2.HTTPBasicAuthHandler(password_mgr)
+        opener = urllib2.build_opener(handler)
+        # Now all calls to urllib2.urlopen use our opener.
+        urllib2.install_opener(opener)
```

Comment:
This is how you can check the output of the main page if it requires a login... don't use firefox.

Reply:
Will update pull request to remove use firefox function.
I can get power device name by you provided method.
Thanks for your help.